src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,225 @@
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.doc;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptEventInfo;
+import ch.njol.skript.lang.SyntaxElement;
+import ch.njol.skript.lang.SyntaxElementInfo;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.skript.lang.function.JavaFunction;
+import ch.njol.skript.registrations.Classes;
+import com.google.common.base.Joiner;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.structure.Structure;
+import org.skriptlang.skript.lang.structure.StructureInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+public class JSONGenerator extends Generator {
+
+	public JSONGenerator(File templateDir, File outputDir) {
+		super(templateDir, outputDir);
+	}
+
+	private static JsonArray convertToJsonArray(String @Nullable[] strings) {
+		if (strings == null)
+			return null;
+
+		JsonArray jsonArray = new JsonArray();
+		for (String string : strings)
+			jsonArray.add(new JsonPrimitive(string));
+		return jsonArray;
+	}
+
+	private static @Nullable String joinStringArray(String @Nullable[] strings, char joiner) {
+		if (strings == null)
+			return null;
+		return Joiner.on(joiner).join(strings);
+	}
+
+	private static @Nullable JsonObject generatedAnnotatedElement(Class<?> syntaxClass, String[] patterns) {
+		if (syntaxClass.getAnnotation(NoDoc.class) != null)
+			return null;
+
+		JsonObject syntaxJsonObject = new JsonObject();
+
+		// TODO: conditionJson.addProperty("id");
+		Name nameAnnotation = syntaxClass.getAnnotation(Name.class);
+		syntaxJsonObject.addProperty("name", nameAnnotation.value());
+
+		Since sinceAnnotation = syntaxClass.getAnnotation(Since.class);
+		syntaxJsonObject.addProperty("since", sinceAnnotation.value());
+
+		Description descriptionAnnotation = syntaxClass.getAnnotation(Description.class);
+		String description = null;
+		if (descriptionAnnotation != null)
+			description = joinStringArray(descriptionAnnotation.value(), '\n');
```

Comment:
What do you think about making all array return types actually be a JSON Array rather than using `\n` joiner? this would be better for consistency.
![image](https://github.com/SkriptLang/Skript/assets/20037329/22875b69-e837-445e-95d0-5eee9332ca4a)

Reply:
i'm not sure. i did it this way because that is how the existing file does it. what do you think? keep or change?

Reply:
I would say change it to actually follow json lists instead of `\n`